keyboards/knops/mini/keymaps/mverteuil/keymap.c

```diff
@@ -0,0 +1,349 @@
+    writePinHigh(D5);
+    writePinLow(B6);
+    writePinHigh(B0);
+
+    if (layerState) {
+        switch (layerId) {
+            case 0:
+                writePinLow(D5);
+                break;
+            case 1:
+                writePinHigh(B6);
+                break;
+            case 2:
+                writePinLow(B0);
+                break;
+            case 3:
+                writePinHigh(D5);
+                writePinLow(B6);
+                writePinHigh(B0);
+                break;
+        }
+    } else {
+        switch (layerId) {
+            case 0:
+                writePinHigh(D5);
+                break;
+            case 1:
+                writePinLow(B6);
+                break;
+            case 2:
+                writePinHigh(B0);
+                break;
+            case 3:
+                writePinLow(D5);
+                writePinHigh(B6);
+                writePinLow(B0);
+                break;
+        }
+    }
+}
+
+/*
+ * This function led_set_layer gets called when you switch between layers.
+ * It allows you to turn on and off leds for each different layer and do
+ * other cool stuff.
+ */
+void led_set_layer(int layer) {
+    switch (layer) {
+        case 0:
+            set_switch_led(1, true);
+            set_switch_led(2, true);
+            set_switch_led(3, true);
+            set_switch_led(4, true);
+            set_switch_led(5, false);
+            set_switch_led(6, false);
+            break;
+
+        case 1:
+            set_switch_led(1, true);
+            set_switch_led(2, true);
+            set_switch_led(3, true);
+            set_switch_led(4, false);
+            set_switch_led(5, true);
+            set_switch_led(6, false);
+            break;
+
+        case 2:
+            set_switch_led(1, true);
+            set_switch_led(2, true);
+            set_switch_led(3, true);
+            set_switch_led(4, false);
+            set_switch_led(5, false);
+            set_switch_led(6, true);
+            break;
+
+        case 3:
+            set_switch_led(1, false);
+            set_switch_led(2, false);
+            set_switch_led(3, true);
+            set_switch_led(4, true);
+            set_switch_led(5, true);
+            set_switch_led(6, true);
+            break;
+    }
+}
+
+void led_init_ports() {
+    writePinHigh(B7);
+    setPinInputHigh(B7);
+
+    setPinOutput(D7);
+    DDRC |= (1 << 6);
+    DDRC |= (1 << 7);
+    DDRD |= (1 << 4);
+    DDRE |= (1 << 6);
+    DDRB |= (1 << 4);
+    DDRD |= (1 << 6);
+
+    DDRD |= (1 << 5);
+    DDRB |= (1 << 6);
+    DDRB |= (1 << 0);;
+
+    led_set_layer(0);
+    wait_ms(200);
+    led_set_layer(1);
+    _delay_ms(200);
+    led_set_layer(2);
+    _delay_ms(200);
+    led_set_layer(0);
+}
+
+// on layer change, no matter where the change was initiated
+// Then runs keymap's layer change check
+layer_state_t layer_state_set_user(layer_state_t state) {
+    // There is no layer "0", it's always at least 1
```

Comment:
Not entirely true. Layer 0 is the default layer.

Reply:
Yeah I noticed that and revised the comment, not sure why I never see layer 1 off though